pkg/diagnose/runner.go

```diff
@@ -378,6 +378,75 @@ func RunStdOut(w io.Writer, diagCfg diagnosis.Config, deps SuitesDeps) error {
 
 	if err != nil {
 		fmt.Fprintln(w, color.RedString(fmt.Sprintf("Error running diagnose: %s", err)))
+		return nil, err
+	}
+
+	return diagnoses, nil
+}
+
+// RunJSON enumerate registered Diagnose suites and get their diagnoses
+// for human consumption in json format output
+func RunJSON(w io.Writer, diagCfg diagnosis.Config, deps SuitesDeps) error {
+	/* Define JSON output structure in the shape of:
+	{
+		"<category name>" : {
+			"<diagnose name>": {
+			diagnosis: string
+			statusCode: int
+			isOk: bool
+			},
+		},
+	}
+	*/
+	// Diagnosis represents the innermost data structure
+	type diagnosis struct {
+		Diagnosis  string `json:"diagnosis"`
+		StatusCode int    `json:"statusCode"`
+		IsOk       bool   `json:"isOk"`
+	}
+	// Category maps a diagnosis name to its details
+	type categoryName map[string]diagnosis
+	// JSONOutput maps a category name to its details
+	type JSONOutput map[string]categoryName
+
+	diagnose, err := runDiagnose(w, diagCfg, deps)
+	if err != nil {
+		return err
```

Comment:
**❓ question:** ‏are we sure that `err` is JSON formatted here ? (e.g. if I have an error when I run `--json` I might want a JSON output `{"error": "..."}`)

Reply:
No this not json formatted but I can do it

Reply:
I tried something but not sure it will work I let you check